processor/transformer/transformer.go

```diff
@@ -296,72 +301,46 @@ func (trans *HandleT) request(ctx context.Context, url string, data []Transforme
 	if err != nil {
 		panic(err)
 	}
-	retryCount := 0
-	var resp *http.Response
-	var respData []byte
-	// We should rarely have error communicating with our JS
-	reqFailed := false
 
 	if len(data) == 0 {
 		return nil
 	}
 
-	// assume that the first event is representative
-
-	for {
-		s := time.Now()
-		trace.WithRegion(ctx, "request/post", func() {
-			resp, err = trans.Client.Post(url, "application/json; charset=utf-8", bytes.NewBuffer(rawJSON))
-		})
-		if err == nil {
-			// If no err returned by client.Post, reading body.
-			// If reading body fails, retrying.
-			respData, err = io.ReadAll(resp.Body)
-			resp.Body.Close()
-		}
-
-		if err != nil {
-			trans.requestTime(statsTags(data[0]), time.Since(s))
-			reqFailed = true
-			trans.logger.Errorf("JS HTTP connection error: URL: %v Error: %+v", url, err)
-			if retryCount > maxRetry {
-				panic(fmt.Errorf("JS HTTP connection error: URL: %v Error: %+v", url, err))
-			}
-			retryCount++
-			time.Sleep(retrySleep)
-			// Refresh the connection
-			continue
-		}
-		if reqFailed {
-			trans.logger.Errorf("Failed request succeeded after %v retries, URL: %v", retryCount, url)
-		}
+	var (
+		respData   []byte
+		statusCode int
+	)
 
-		// perform version compatibility check only on success
-		if resp.StatusCode == http.StatusOK {
-			transformerAPIVersion, convErr := strconv.Atoi(resp.Header.Get("apiVersion"))
-			if convErr != nil {
-				transformerAPIVersion = 0
+	// endless retry if transformer-controlplane connection is down
+	endlessBackoff := backoff.NewExponentialBackOff()
+	endlessBackoff.MaxElapsedTime = 0 // no max time -> ends only when no error
+	// endless backoff loop, only nil error or panics inside
+	_ = backoff.RetryNotify(
+		func() error {
+			respData, statusCode = trans.doPost(ctx, rawJSON, url, statsTags(data[0]))
```

Comment:
**Note:** we need to make sure that transformer uses this status code in its response and doesn't "hide" it inside the json response, masking it in a HTTP 200 response, i.e. using the `statusCode` field of transformed messages.

This functionality also needs to be tested using the relevant transformer release that includes the new status code.